lib/auth/accesslist/service.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package accesslist
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	accesslistv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/accesslist/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/authz"
+	"github.com/gravitational/teleport/lib/backend"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/services/local"
+	conv "github.com/gravitational/teleport/lib/types/accesslist/convert/v1"
+)
+
+// ServiceConfig is the service config for the Access Lists gRPC service.
+type ServiceConfig struct {
+	// Backend is the backend to use.
+	Backend backend.Backend
+
+	// Logger is the logger to use.
+	Logger logrus.FieldLogger
+
+	// Authorizer is the authorizer to use.
+	Authorizer authz.Authorizer
+
+	// AccessLists is the access list service to use.
+	AccessLists services.AccessLists
+}
+
+func (c *ServiceConfig) validateConfig() error {
```

Comment:
Usually, we name these functions as `CheckAndSetDefaults` .

Since you assign the logger I think it's worth renaming

Reply:
oh ok. If @codingllama  asks for it, @codingllama  will have it. :smile: 
It's just because it's confusing that a `validateConfig` function sets some fields if they don't exist.